17-lifecycle-lab/src/04-redspotter.jsx

```diff
@@ -1,48 +1,77 @@
-/**
- * Add a timer to the following Red Spotter game
- * When a user doesn't do anything for 2 seconds, she should
- * lose 3 points.
- */
-
-import React from 'react';
-import _ from 'underscore';
-
-const SQUARES_COUNT = 10;
-const SQUARE_STYLE = {
-  width: '100px',
-  height: '100px',
-  background: 'gray',
-  display: 'inline-block',
-};
-
-const WINNER_STYLE = Object.assign({}, SQUARE_STYLE, {
-  background: 'red',
-});
-
-export default React.createClass({
-  getInitialState() {
-    return {
-      winner: _.random(SQUARES_COUNT),
-      score: 0,
-    };
-  },
-
-  shuffle() {
-    this.setState({ 
-      winner: _.random(SQUARES_COUNT),
-      score: this.state.score + 5,
-    });
-  },
-
-  render() {
-    return (<div>
-      <p>Score: {this.state.score}</p>
-      {_.range(SQUARES_COUNT).map((index) => (
-        <div 
-          onClick={() => (this.state.winner === index) && this.shuffle()}
-          style={index === this.state.winner ? WINNER_STYLE : SQUARE_STYLE} />
-
-      ))}
-    </div>);
-  }
-});
+/**
+ * Add a timer to the following Red Spotter game
+ * When a user doesn't do anything for 2 seconds, she should
+ * lose 3 points.
+ */
+
+import React from 'react';
+
+import RedButton from '04-redbutton';
+
+export default class RedSpotter extends React.Component {
+    constructor(props) {
+        super(props);
+        this.state = {
+            redLocation: 0,
+            score: 0,
+        };
+    }
+
+    restart = () => {
+      this.state.score = 0;
+      this.reOrderButtons();
+    }
+
+    playGame = (clickedColor) => {
+        this.resetTimer();
+        console.log("clicked color " + clickedColor);
+        if (clickedColor === 'red') {

```

Comment:
Encapsulate the whole if inside another method. Call it applyScoreByColor(color)